### PR TITLE
feat: ability to send headers when connect to browser using puppeteer

### PR DESCRIPTION
**Pull request by DudaGod**
## Proposed changes

Support changes from https://github.com/webdriverio/webdriverio/pull/9354 in wdio@7.

### CODE REVIEW

**Comment by christian-bromann** [comment]
> chore: bump puppeteer-core from 13.1.3 to 19.4.0

This caused some type issues due to file refactoring on the Protractor side. You can resolve them checking into the `main` branch. Let me know if you have any questions.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
Seems that there are still build issues:

> Error: ../../../packages/devtools/node_modules/puppeteer-core/lib/types.d.ts(33,5): error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

**Comment by DudaGod** [comment]
> Seems that there are still build issues:

Yeah, I see. I will ping you when all the tests pass successfully.

**Comment by DudaGod** [comment]
I finally fixed everything. Can you review it now?

File: e2e/wdio/headless/test.e2e.js [hunk 1]
```diff
@@ -8,15 +8,27 @@ describe('main suite 1', () => {
...
         await browser.url('https://json.org')
         expect((await browser.checkPWA()).passed).toBe(false)
     })
 
-    it('should allow to do performance tests', async () => {
+    /**
+     * fails due to "Unable to identify the main resource"
+     * https://github.com/webdriverio/webdriverio/issues/8541
+     */
+    it.skip('should allow to do performance tests', async () => {
```
Comment (christian-bromann) [inline e2e/wdio/headless/test.e2e.js:31]: Is this failing for `v7` as well?

**Review by christian-bromann — COMMENTED** [review]
Great work 👍  one question

File: e2e/wdio/headless/test.e2e.js (hunk 1 above)
Comment (DudaGod) [reply e2e/wdio/headless/test.e2e.js:31]: Yes, with the same error. So I found that this test skipped in v8 and skip it as well for v7.

Comment (christian-bromann) [reply e2e/wdio/headless/test.e2e.js:31]: 👍

**Review by christian-bromann — APPROVED** [review]
Thanks 👍

**Comment by DudaGod** [comment]
@christian-bromann can you publish new wdio@7 version to npm? Thank you.

**Comment by kyryloonufriiev** [comment]
@DudaGod This version could not be built with ts

**Comment by DudaGod** [comment]
> This version could not be built with ts

Are you talking about webdriverio@7.28.0 - https://github.com/webdriverio/webdriverio/releases/tag/v7.28.0 ?
What particular error are you getting?

**Comment by christian-bromann** [comment]
@DudaGod I had to revert this change due to the fact that it breaks for folks using Node.js v12 and v13. Maybe this change can be applied without updating Puppeteer.

**Comment by DudaGod** [comment]
> I had to revert this change due to the fact that it breaks for folks using Node.js v12 and v13. 

Sorry, I did not notice that the required version of node.js has changed in puppeteer-core.

> Maybe this change can be applied without updating Puppeteer.

Only if I send a new PR for puppeteer-core@14, but 5 majors have already been released, so they are unlikely to agree to merge my changes. I will try to move to wdio@8

**Pull request merged**